### PR TITLE
Add section on configuring OnPremise system dependencies

### PR DESCRIPTION
**Pull request by bpostlethwaite**
@cldougl 

I think you will be able to see what it looks like in the Percy snapshots... if we test the OnPremise section. If not I can post a screenie here.

### CODE REVIEW

**Comment by bpostlethwaite** [comment]
Yep Percy screeny for the new section is here:
https://percy.io/plotly/dash-docs/builds/79160

1 diff change is curious, Radio items for "canada" and "america" swapped places even though they have nothing to do with the files I modified. I'll take a look.

**Comment by bpostlethwaite** [comment]
Oh there is a View Deployment option for each PR. Nice!

**Comment by chriddyp** [comment]
> Radio items for "canada" and "america" swapped places even though they have nothing to do with the files I modified


don’t worry about this - the example uses dicts which are unordered and occasionally swap places

File: tutorial/on_premise_deployment.py [hunk 1]
```diff
@@ -338,6 +341,82 @@ def generate_instructions(chapter, platform):
...
+            unixodbc-dev
+            '''), customStyle=styles.code_container, language="text"),
+
+            dcc.Markdown(s('''
+
+            #### Configure System Dependencies
+
+            You may include a pre-deploy script that executes in
+            your Dash App's environment. For the case of adding an
+            ODBC driver we may want to copy ODBC initialization files
```
Comment (cldougl) [inline tutorial/on_premise_deployment.py]: I would probably take at that "may" cause this isn't really an optional step for this case

File: tutorial/on_premise_deployment.py [hunk 2]
```diff
@@ -338,6 +341,82 @@ def generate_instructions(chapter, platform):
...
+            from our application folder into system paths where the
+            ODBC driver expects them to be.
+
+            ##### Add A Pre-Deploy Script
+            Let's generate a file to do this. Note that the file can
+            have any name as we must specify the name in an application
+            configuration file `app.json`.
+            For the purposes of this example we assume we have
+            named it `setup_pyodbc`.
```
Comment (cldougl) [inline tutorial/on_premise_deployment.py]: should mention this is in the root directory of the application

File: tutorial/on_premise_deployment.py [hunk 3]
```diff
@@ -338,6 +341,82 @@ def generate_instructions(chapter, platform):
...
+            '''), customStyle=styles.code_container, language='json'),
+
+            dcc.Markdown(s('''
+            ***
+
+            Now when the application is deployed it will install the apt
+            packages specified in `apt-packages` and run the setup file
+            specified in `app.json`. In this case it allows us to install
+            and then configure the ODBC driver.
+            '''))
```
Comment (cldougl) [inline tutorial/on_premise_deployment.py]: I would add a link to the PR examples [pyodbc example](https://github.com/plotly/dash-on-premise-sample-app/pull/3#issue-144272510) as well (thus far it was useful for most of the users to work off of that example as some of them are unfamiliar with deployment processes)

**Comment by bpostlethwaite** [comment]
@cldougl ok ready for another review thanks!

**Comment by bpostlethwaite** [comment]
Actually looks like you have given me a :dancer:  - ok I'll approve the Percy shots and merge.

**Comment by cldougl** [comment]
yep- nothing blocking :) the changes look good to me 🎉

**Pull request merged**